### PR TITLE
Add LCAO performance tests

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
These benchmarks are VMC and DMC runs for carbon molecules in order to measure LCAO performance for various system sizes. The simulated systems are various sizes of a carbon ring (C12, C18 ,C24, and C30) and C60 buckyball. All-electron tests (ae) with the cusp correction and tests with a pseudopotential (pp) are included in carbon ring tests.

 Name  Atoms    Electrons        Electrons per spin
 C12        12    48(pp), 72(ae)      24(pp),36(ae)
 C18        18    72(pp),108(ae)     36(pp),54(ae)
 C24        24    96(pp),144(ae)     48(pp),72(ae)
 C30        30   120(pp),180(ae)     60(pp),90(ae)
 C60        60            240(pp)            120(pp)

More details (h5 files location, etc.) can be found in README.

Hyeondeok Shin

### CODE REVIEW

**Comment by prckent** [comment]
OK to test

File: tests/performance/C-molecule/README [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+three body Jastrow function. 
+
+The tests require at least QMCPACK 3.6.0.
+
+3. Requirements
+
+Download the necessary C-molecule h5 orbital files of different sizes from
+the following link
+
+https://anl.box.com/s/pveyyzrc2wuvg5tmxjzzwxeo561vh3r0
```
Comment (ye-luo) [inline tests/performance/C-molecule/README]: This is the correct link.
https://anl.app.box.com/s/88z6quep9e7gp8vqagtupkg73zdumpaj

File: manual/installation.tex [hunk 2]
```diff
@@ -349,17 +349,26 @@ \subsection{Role of QMC\_DATA}
...
 size. The tests are activated if CMake detects the files when
 configured. See tests/performance/NiO/README,
-tests/solids/NiO\_afqmc/README, and tests/performance/C-graphite/README
+tests/solids/NiO\_afqmc/README, tests/performance/C-graphite/README, and tests/performance/C-molecule/README
 for details of the current tests and input files and to download them.
 
 Currently the files must be downloaded via\\
 \url{https://anl.box.com/s/pveyyzrc2wuvg5tmxjzzwxeo561vh3r0} and \\
-\url{https://anl.box.com/s/j5d6jeazpgx5441s04ajtv77iogoyjsh}.
+\url{https://anl.box.com/s/j5d6jeazpgx5441s04ajtv77iogoyjsh} (C-graphite file).
```
Comment (ye-luo) [inline manual/installation.tex]: Please add a link to C h5 files.

**Comment by ye-luo** [comment]
Please remove C.BFD.xml and use softlink instead. There is a copy at `tests/pseudopotentials_for_tests`

**Comment by prckent** [comment]
@ye-luo I changed the link intentionally. We should use a single URL for *all* our files from now on. Please work with @Hyeondeok-Shin to update appropriately.

File: manual/installation.tex (hunk 2 above)
Comment (prckent) [reply manual/installation.tex]: See comment below. After updating we should only have a single unique box url.

File: tests/performance/C-molecule/README (hunk 1 above)
Comment (prckent) [reply tests/performance/C-molecule/README]: Change of strategy. Single URL for all files from now on.

**Comment by prckent** [comment]
Confirming that these tests run OK (oxygen, PGI2019 compiler with MKL). We might consider more steps per block in future since the smaller systems are perhaps too fast. There are some interesting timing differences between ae and pp runs, but we should check more run, compilers & platforms before drawing conclusions.

```
$ grep molec ctest.out
      Start  88: performance-C-molecule-C12-e72-ae-1-16
39/74 Test  #88: performance-C-molecule-C12-e72-ae-1-16 ....................   Passed    1.38 sec
      Start  89: performance-C-molecule-C12-e72-ae-1-16-time
40/74 Test  #89: performance-C-molecule-C12-e72-ae-1-16-time ...............   Passed    0.02 sec
      Start  90: performance-C-molecule-C12-e72-ae-J3-1-16
41/74 Test  #90: performance-C-molecule-C12-e72-ae-J3-1-16 .................   Passed    1.54 sec
      Start  91: performance-C-molecule-C12-e72-ae-J3-1-16-time
42/74 Test  #91: performance-C-molecule-C12-e72-ae-J3-1-16-time ............   Passed    0.02 sec
      Start  92: performance-C-molecule-C12-e48-pp-1-16
43/74 Test  #92: performance-C-molecule-C12-e48-pp-1-16 ....................   Passed    0.37 sec
      Start  93: performance-C-molecule-C12-e48-pp-1-16-time
44/74 Test  #93: performance-C-molecule-C12-e48-pp-1-16-time ...............   Passed    0.02 sec
      Start  94: performance-C-molecule-C12-e48-pp-J3-1-16
45/74 Test  #94: performance-C-molecule-C12-e48-pp-J3-1-16 .................   Passed    0.45 sec
      Start  95: performance-C-molecule-C12-e48-pp-J3-1-16-time
46/74 Test  #95: performance-C-molecule-C12-e48-pp-J3-1-16-time ............   Passed    0.03 sec
      Start  96: performance-C-molecule-C18-e108-ae-1-16
47/74 Test  #96: performance-C-molecule-C18-e108-ae-1-16 ...................   Passed    4.33 sec
      Start  97: performance-C-molecule-C18-e108-ae-1-16-time
48/74 Test  #97: performance-C-molecule-C18-e108-ae-1-16-time ..............   Passed    0.02 sec
      Start  98: performance-C-molecule-C18-e108-ae-J3-1-16
49/74 Test  #98: performance-C-molecule-C18-e108-ae-J3-1-16 ................   Passed    4.58 sec
      Start  99: performance-C-molecule-C18-e108-ae-J3-1-16-time
50/74 Test  #99: performance-C-molecule-C18-e108-ae-J3-1-16-time ...........   Passed    0.02 sec
      Start 100: performance-C-molecule-C18-e72-pp-1-16
51/74 Test #100: performance-C-molecule-C18-e72-pp-1-16 ....................   Passed    0.64 sec
      Start 101: performance-C-molecule-C18-e72-pp-1-16-time
52/74 Test #101: performance-C-molecule-C18-e72-pp-1-16-time ...............   Passed    0.02 sec
      Start 102: performance-C-molecule-C18-e72-pp-J3-1-16
53/74 Test #102: performance-C-molecule-C18-e72-pp-J3-1-16 .................   Passed    0.78 sec
      Start 103: performance-C-molecule-C18-e72-pp-J3-1-16-time
54/74 Test #103: performance-C-molecule-C18-e72-pp-J3-1-16-time ............   Passed    0.02 sec
      Start 104: performance-C-molecule-C24-e144-ae-1-16
55/74 Test #104: performance-C-molecule-C24-e144-ae-1-16 ...................   Passed   12.22 sec
      Start 105: performance-C-molecule-C24-e144-ae-1-16-time
56/74 Test #105: performance-C-molecule-C24-e144-ae-1-16-time ..............   Passed    0.02 sec
      Start 106: performance-C-molecule-C24-e144-ae-J3-1-16
57/74 Test #106: performance-C-molecule-C24-e144-ae-J3-1-16 ................   Passed   11.97 sec
      Start 107: performance-C-molecule-C24-e144-ae-J3-1-16-time
58/74 Test #107: performance-C-molecule-C24-e144-ae-J3-1-16-time ...........   Passed    0.02 sec
      Start 108: performance-C-molecule-C24-e96-pp-1-16
59/74 Test #108: performance-C-molecule-C24-e96-pp-1-16 ....................   Passed    1.08 sec
      Start 109: performance-C-molecule-C24-e96-pp-1-16-time
60/74 Test #109: performance-C-molecule-C24-e96-pp-1-16-time ...............   Passed    0.02 sec
      Start 110: performance-C-molecule-C24-e96-pp-J3-1-16
61/74 Test #110: performance-C-molecule-C24-e96-pp-J3-1-16 .................   Passed    1.30 sec
      Start 111: performance-C-molecule-C24-e96-pp-J3-1-16-time
62/74 Test #111: performance-C-molecule-C24-e96-pp-J3-1-16-time ............   Passed    0.02 sec
      Start 112: performance-C-molecule-C30-e180-ae-1-16
63/74 Test #112: performance-C-molecule-C30-e180-ae-1-16 ...................   Passed   28.05 sec
      Start 113: performance-C-molecule-C30-e180-ae-1-16-time
64/74 Test #113: performance-C-molecule-C30-e180-ae-1-16-time ..............   Passed    0.03 sec
      Start 114: performance-C-molecule-C30-e180-ae-J3-1-16
65/74 Test #114: performance-C-molecule-C30-e180-ae-J3-1-16 ................   Passed   28.94 sec
      Start 115: performance-C-molecule-C30-e180-ae-J3-1-16-time
66/74 Test #115: performance-C-molecule-C30-e180-ae-J3-1-16-time ...........   Passed    0.03 sec
      Start 116: performance-C-molecule-C30-e120-pp-1-16
67/74 Test #116: performance-C-molecule-C30-e120-pp-1-16 ...................   Passed    1.71 sec
      Start 117: performance-C-molecule-C30-e120-pp-1-16-time
68/74 Test #117: performance-C-molecule-C30-e120-pp-1-16-time ..............   Passed    0.02 sec
      Start 118: performance-C-molecule-C30-e120-pp-J3-1-16
69/74 Test #118: performance-C-molecule-C30-e120-pp-J3-1-16 ................   Passed    1.95 sec
      Start 119: performance-C-molecule-C30-e120-pp-J3-1-16-time
70/74 Test #119: performance-C-molecule-C30-e120-pp-J3-1-16-time ...........   Passed    0.03 sec
      Start 120: performance-C-molecule-C60-e240-pp-1-16
71/74 Test #120: performance-C-molecule-C60-e240-pp-1-16 ...................   Passed   11.00 sec
      Start 121: performance-C-molecule-C60-e240-pp-1-16-time
72/74 Test #121: performance-C-molecule-C60-e240-pp-1-16-time ..............   Passed    0.03 sec
      Start 122: performance-C-molecule-C60-e240-pp-J3-1-16
73/74 Test #122: performance-C-molecule-C60-e240-pp-J3-1-16 ................   Passed   11.80 sec
      Start 123: performance-C-molecule-C60-e240-pp-J3-1-16-time
74/74 Test #123: performance-C-molecule-C60-e240-pp-J3-1-16-time ...........   Passed    0.03 sec

```

**Comment by Hyeondeok-Shin** [comment]
@prckent @ye-luo Now box URLs for h5 are replaced to the same upper folder for all performance tests. H5 file located sub-folders are using the name of a system in the tests, so users can easily find h5 files. 
 As @ye-luo pointed out, now a carbon PP is using the softlink. 
 Current C-molecule is using exactly the same # of steps and blocks with the NiO performance tests. I'll keep watching the performance tests and figure out how much these steps & blocks should increase if it needs.

**Pull request merged**